index.d.ts

```diff
@@ -1,30 +1,67 @@
+import {type RequireAtLeastOne} from 'type-fest';
+
+export type Frame = {
+	x: number;
+	y: number;
+	width: number;
+	height: number;
+};
+
 export type Screen = {
-	id: number;
+	id: string;
 	name: string;
+	width: number;
+	height: number;
+	frame: Frame;
+};
+
+export type Window = {
+	id: string;
+	title?: string;
+	applicationName?: string;
+	applicationBundleIdentifier?: string;
```

Comment:
Nitpick (my preference only):

```suggestion
	appName?: string;
	appBundleIdentifier?: string;
```